tools/ci/checks/run_flake8_checks.sh

```diff
@@ -1,6 +1,11 @@
 #!/bin/bash
 # Run flake8 in a docker container.
 
-TOOL_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+DOCKER_TAG=${DOCKER_TAG:-latest}
+DOCKER_IMAGE=${DOCKER_IMAGE:-mesosphere/dcos-commons:${DOCKER_TAG}}
 
-${TOOL_DIR}/run_check_in_docker.sh flake8 $*
+docker run --rm -t \
+    -v $(pwd):/build:ro \
+    -w /build \
+        ${DOCKER_IMAGE} \
+            flake8 $@
```

Comment:
It's a good idea to always quote variables. Maybe we can get ShellCheck running against our bash in the near future too...

Reply:
They do behave differently though. In both cases, there is a difference between passing the arguments as a space separated list (`$@`) and a quoted mulit-line


In the case of `pylint`:
```
+ docker run --rm -t -v /Users/elezar/src/mesosphere/dcos-commons:/build:ro -w /build mesosphere/dcos-commons:latest pylint -E testing/sdk_diag.py testing/sdk_hosts.py testing/sdk_install.py testing/sdk_jobs.py testing/sdk_metrics.py testing/sdk_plan.py testing/sdk_security.py testing/sdk_tasks.py testing/sdk_upgrade.py tools/ci/checks/get_applicable_changes.py
```
vs
```
docker run --rm -t -v /Users/elezar/src/mesosphere/dcos-commons:/build:ro -w /build mesosphere/dcos-commons:latest pylint -E 'testing/sdk_diag.py
testing/sdk_hosts.py
testing/sdk_install.py
testing/sdk_jobs.py
testing/sdk_metrics.py
testing/sdk_plan.py
testing/sdk_security.py
testing/sdk_tasks.py
testing/sdk_upgrade.py
tools/ci/checks/get_applicable_changes.py'
```

From http://tldp.org/LDP/abs/html/internalvariables.html
> Same as $*, but each parameter is a quoted string, that is, the parameters are passed on intact, without interpretation or expansion. This means, among other things, that each parameter in the argument list is seen as a separate word.

Which is what I want since each element of `$@` is a separate filename.

Reply:
Right, this is the point. Please pass them as individual elements instead. Otherwise we'll be in a world of pain as soon as a file/dir name with space appears.

Reply:
I don't quite follow what you mean. What I'm saying is that `$@` already quotes the separate elements and `"$@"` does *not* work for these scripts.

Reply:
Here's a test case:
```
porridge@beczulka:~/tmp/x$ mkdir t
porridge@beczulka:~/tmp/x$ touch t/1 't/2 2'
porridge@beczulka:~/tmp/x$ echo 'ls -l "$@"' > quoted
porridge@beczulka:~/tmp/x$ echo 'ls -l $@' > unquoted
porridge@beczulka:~/tmp/x$ bash quoted t/*
-rw-r--r-- 1 porridge porridge 0 lip  2 18:18 t/1
-rw-r--r-- 1 porridge porridge 0 lip  2 18:18 t/2 2
porridge@beczulka:~/tmp/x$ bash unquoted t/*
ls: nie ma dostępu do 't/2': Nie ma takiego pliku ani katalogu
ls: nie ma dostępu do '2': Nie ma takiego pliku ani katalogu
-rw-r--r-- 1 porridge porridge 0 lip  2 18:18 t/1
porridge@beczulka:~/tmp/x$ bash quoted "t/1 t/2 2"
ls: nie ma dostępu do 't/1 t/2 2': Nie ma takiego pliku ani katalogu
porridge@beczulka:~/tmp/x$ bash unquoted "t/1 t/2 2"
ls: nie ma dostępu do 't/2': Nie ma takiego pliku ani katalogu
ls: nie ma dostępu do '2': Nie ma takiego pliku ani katalogu
-rw-r--r-- 1 porridge porridge 0 lip  2 18:18 t/1
porridge@beczulka:~/tmp/x$ 
```